src/main/java/ecommerce/products/controllers/ProductController.java

```diff
@@ -0,0 +1,73 @@
+package ecommerce.products.controllers;
+
+import ecommerce.products.models.Product;
+import ecommerce.products.repository.ProductRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("api/products")
+public class ProductController {
+
+    @Autowired
```

Comment:
`@Autowired` is deprecated is no longer used. Replace it with `@RequiredArgsConstructor`  form `Lombok`.

eg.
```java

@RestController
@RequestMapping("api/products")
@RequiredArgsConstructor
public class ProductController {
```

Reply:
solved